var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -254,8 +254,8 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     conflicts('+bootstrap', when='@:11.1 %apple-clang@12.0.5')
 
     # aarch64/M1 is supported in GCC 12+
-    conflicts('@:11', when='target=aarch64: platform=darwin',
-              msg='Only GCC 12 and newer support macOS M1 (aarch64)')
+    conflicts('@:11.2,11.3.1:', when='target=aarch64: platform=darwin',
+              msg='Only GCC 11.3.0 supports macOS M1 (aarch64)')
```

Comment:
GCC 12 was supposed to have M1 support but that feature was delayed. As far as I know, the master branch still doesn't have M1 support, so until that happens we should restrict it to the patched 11.3.0 version.